### PR TITLE
binary expression should simplify when either operant is null

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

binary expression should simplify when either operant is null

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by houqp — DISMISSED** [review]

File: datafusion/src/optimizer/simplify_expressions.rs [hunk 1]
```diff
@@ -100,6 +100,8 @@ fn is_false(expr: &Expr) -> bool {
 
 fn simplify(expr: &Expr) -> Expr {
     match expr {
+        Expr::BinaryExpr { left, .. } if is_literal_null(left) => *left.clone(),
```
Comment (alamb) [inline datafusion/src/optimizer/simplify_expressions.rs:103]: Thank you @Jimexist

I don't think this is true for `Operator::IsDistinctFrom` and `Operator::IsNotDistinctFrom` ,   `Operator::And` and `Operator::Or`,  so I think we should special case those Operators

Or perhaps to be more conservative make a whitelist of the arithmetic operators.


https://github.com/apache/arrow-datafusion/blob/81fae230b81b97e93bbb95b284cfda6f4d59552e/datafusion/src/logical_plan/operators.rs#L56-L58

examples

`null IS DISTINCT FROM null` --> `false` (but this PR would rewrite to `null` I think)

`true OR null` --> `true` (but this PR would rewrite it to `null` I think)

File: datafusion/src/optimizer/simplify_expressions.rs [hunk 2]
```diff
@@ -453,7 +450,7 @@ mod tests {
     fn test_simplify_and_and_false() -> Result<()> {
         let expr =
             binary_expr(lit(ScalarValue::Boolean(None)), Operator::And, lit(false));
-        let expr_eq = lit(false);
+        let expr_eq = null;
```
Comment (alamb) [inline datafusion/src/optimizer/simplify_expressions.rs:453]: I think the original answer of `false` is correct. Here is the result in postgres:

```
alamb=# select null and false;
 ?column? 
----------
 f
(1 row)

alamb=# 
```

**Comment by alamb** [comment]
I am trying to clean up the list of PRs to review in DataFusion so marking old ones as stale. Please let us know if you plan to work on this soon. Otherwise we will close it down and reopen it when the time is right.

**Comment by alamb** [comment]
Closing stale PRs, please reopen if this was a mistake and you plan to keep working on this one

**Pull request closed without merging**